### PR TITLE
improve(serverless-hub): Update spoke error log

### LINKED ISSUE #2112: Improve hub logging output details
**Why**

Right now when an error is generated in a spoke the hub simply informs slack that "something went wrong" in the following spokes. For example, the screenshot below shows an output of one bot with error, 5 without error:
image

This is helpful, but not very informative. Ideally, the hub logs should also show the error message that was generated.

**How**

Update the hub implementation to show the exact error generated with each spoke that failed. This should still be clean and easy to read in slack.

**Resolution**

Slack logs are improved. Tests should also be updated.

**Difficulty Score [1-10]**

- 6: ~1 day

### PR DESCRIPTION
**Pull request by nicholaspai**
**Motivation**

#2112

**Summary**

This PR refactors how the hub deals with error messages by making two main changes:
1) the hub now detects if the error thrown occurred in the hub process (an "internal hub error") and throws accordingly.
2) if the error was created in a spoke process then the hub will print the associated error message with the spoke process that generated it.

**Details**

**Issue(s)**

Fixes #2112

### CODE REVIEW

File: packages/core/scripts/serverless-orchestration/ServerlessHub.js
Comment (nicholaspai) [inline packages/core/scripts/serverless-orchestration/ServerlessHub.js:177]: Ideally I would add `errorMessages` under `output` like int he `logger.error(...)` above, but I didn't want to mess up the shape of the `output` key in this server response because I do not know what is downstream of this.

Also, curious if this should be renamed to `spokeErrorMessages` for example, since only errors originating from the spokes will appear here. If the error comes from the Hub code, then there will not be any `errorMessages`

File: packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js
Comment (chrismaree) [inline packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:172]: Good catch @nicholaspai. I've added an issue to track this: https://github.com/UMAprotocol/protocol/issues/2121

File: packages/core/scripts/serverless-orchestration/ServerlessHub.js
Comment (nicholaspai) [inline packages/core/scripts/serverless-orchestration/ServerlessHub.js:144]: +1

File: packages/core/scripts/serverless-orchestration/ServerlessHub.js
Comment (nicholaspai) [inline packages/core/scripts/serverless-orchestration/ServerlessHub.js:163]: Should we worry about edge case where `errorOutput.errorOutputs[spokeName]` is larger than SlackTransport can handle?

File: packages/core/scripts/serverless-orchestration/test/ServerlessHub.js
Comment (nicholaspai) [inline packages/core/scripts/serverless-orchestration/test/ServerlessHub.js:23]: Is this `assert` meant to be here?

File: packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js
Comment (nicholaspai) [reply packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:172]: Yeah we might want to catch these up stream in the SlackTransport for example. It could just truncate messages to the limit.

**Review by nicholaspai — COMMENTED** [review]
I can't approve this because its my own PR but this looks awesome and your work has much refined mine! LGTM but please review my comments

File: packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js
Comment (chrismaree) [reply packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js:172]: ye I agree 100%. it should happen in the transport.

File: packages/core/scripts/serverless-orchestration/ServerlessHub.js
Comment (chrismaree) [reply packages/core/scripts/serverless-orchestration/ServerlessHub.js:163]: I think the SlackTransport should handle this. Can leave it for https://github.com/UMAprotocol/protocol/issues/2121

**Review by chrismaree — APPROVED** [review]
looks like a clean refactor and improvement! good work @nicholaspai

File: packages/core/scripts/serverless-orchestration/test/ServerlessHub.js
Comment (chrismaree) [reply packages/core/scripts/serverless-orchestration/test/ServerlessHub.js]: nope. I removed it

**Review by daywiss — APPROVED** [review]

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**
